vcd/resource_vcd_service_account.go

```diff
@@ -0,0 +1,344 @@
+				Required:    true,
+				Description: "Any valid UUID, depends on the user, e.g: 12345678-1234-5678-90ab-1234567890ab",
+			},
+			"role_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Role ID of service account",
+			},
+			"software_version": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Version of software using the service account, can be freely defined by the user",
+			},
+			"uri": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "URI of the client using the service account, can be freely defined by the user",
+			},
+			"active": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Status of the service account.",
+			},
+			"file_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Name of the file that the API token will be saved to",
+			},
+			"allow_token_file": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+				Description: "Set this to true if you understand the security risks of using" +
+					" API token files and would like to suppress the warnings",
+			},
+		},
+	}
+}
+
+func resourceVcdServiceAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	vcdClient := meta.(*VCDClient)
+
+	org, err := vcdClient.GetOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Service Account create] error getting Org from resource: %s", err)
+	}
+
+	adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+	if err != nil {
+		return diag.Errorf("[Service Account create] error getting AdminOrg from resource: %s", err)
+	}
+
+	saName := d.Get("name").(string)
+	softwareId := d.Get("software_id").(string)
+	softwareVersion := d.Get("software_version").(string)
+	uri := d.Get("uri").(string)
+	filename := d.Get("file_name").(string)
+
+	// Role needs to be sent in URN format, and the role name needs to be percent-encoded
+	// e.g urn:vcloud:role:Organization%20Administrator
+	roleId := d.Get("role_id").(string)
+	role, err := adminOrg.GetRoleById(roleId)
+	if err != nil {
+		return diag.Errorf("[Service Account create] error getting Role: %s", err)
+	}
+	escapedRoleName := url.PathEscape(role.Role.Name)
+	formattedRole := "urn:vcloud:role:" + escapedRoleName
+
+	active := d.Get("active").(bool)
+	allowTokenFile := d.Get("allow_token_file").(bool)
+
+	if active && !allowTokenFile {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "The file " + filename + " should be considered sensitive information.",
+				Detail: "The file " + filename + " will contain the initial service account API " +
+					"and should be considered SENSITIVE INFORMATION. " +
+					"The token will be invalidated at the next use but a new one will be provided in the request's response. If you acknowledge this, add\n\n	allow_token_file = true\n\nto the provider settings.",
+			},
+		}
+	}
```

Comment:
There is a case where I can put:
```
  active = true
  allow_token_file = true
```

But skip the `file_name` - then the message is a bit messy:
```
│ Error: The file  should be considered sensitive information.
│ 
│   with vcd_service_account.sysadmin,
│   on vcd.TestAccServiceAccount_SysOrgstep1.tf line 32, in resource "vcd_service_account" "sysadmin":
│   32: resource "vcd_service_account" "sysadmin" {
│ 
│ The file  will contain the initial service account API and should be considered SENSITIVE INFORMATION. The token will be invalidated at the next use but a new one
│ will be provided in the request's response. If you acknowledge this, add
│ 
│       allow_token_file = true
│ 
│ to the provider settings.
```

Many ways to approach it, but maybe there could be a separate validation before this one to check that if `allow_token_file=true`, but `file_name` is empty.

Reply:
Added a check before that, similar to `Update`, so it returns an error `if active && filename == ""`